src/crepe/api/Camera.h

```diff
@@ -20,33 +21,35 @@ class Camera : public Component {
 	 * \param id Unique identifier for the camera component.
 	 * \param bg_color Background color for the camera view.
 	 */
-	Camera(game_object_id_t id, const Color & bg_color);
+	Camera(game_object_id_t id, const Color & bg_color, const ivec2 & screen,
+		   const ivec2 & viewport, const double & zoom, const vec2 & offset = {0, 0});
 	~Camera(); // dbg_trace only
 
 public:
 	//! Background color of the camera view.
-	Color bg_color;
+	const Color bg_color;
 
-	//! Aspect ratio height for the camera.
-	double aspect_height = 480;
+	//! offset postion from the game object transform component
+	vec2 offset;
 
-	//! Aspect ratio width for the camera.
-	double aspect_width = 640;
+	//! pos the postion of the camera in world space this will be filled with
+	//pos = transform + offset
```

Comment:
Klein ding tussendoor vec2 pos = {0,0} kan gewoon vec2 pos; zijn vec2 is standaard 0,0 zoals je wel bij offset hebt gedaan hierboven

Reply:
daar ga ik niet op vertrouwen

Reply:
daar hoef je niet op te vertrouwen want dit doe je al overal. soms heb je het al gedaan zonder. en het is niet vertrouwen zie de vec2 klasse